client/src/jobs/job.py

```diff
@@ -66,6 +56,23 @@ class ImageOptions(BaseModel):
         protected_namespaces=(),
     )
 
+    def __init__(self, **data: Any) -> None:
```

Comment:
Since this is a Pydantic model, wouldn't it be more idiomatic to perform the check in a field validator? That way we wouldn't have to mess with the constructor.